pkg/provisioner/templates/containerd.go

```diff
@@ -32,29 +32,121 @@ const containerdTemplate = `
+
+# Download containerd tarball and checksum
+curl -fsSL -o ${CONTAINERD_TAR} ${CONTAINERD_URL}
+curl -fsSL -o containerd_SHA256SUMS ${CONTAINERD_SHA256_URL}
+
+# Verify SHA256 checksum
+EXPECTED_CHECKSUM=$(grep ${CONTAINERD_TAR} containerd_SHA256SUMS | awk '{print $1}')
+ACTUAL_CHECKSUM=$(sha256sum ${CONTAINERD_TAR} | awk '{print $1}')
+
+if [ "$EXPECTED_CHECKSUM" != "$ACTUAL_CHECKSUM" ]; then
+    echo "Checksum verification failed for containerd. Exiting."
+    exit 1
+fi
+
+# Stream directly into tar to avoid saving the archive
+cat ${CONTAINERD_TAR} | sudo tar Cxzvf /usr/local -
+
+# Cleanup
+rm -f ${CONTAINERD_TAR} containerd_SHA256SUMS
+
+echo "Containerd ${CONTAINERD_VERSION} installed successfully."
+
+# Fetch latest stable RUNC version from GitHub
+echo "Fetching latest stable runc version..."
+RUNC_VERSION=$(curl -fsSL https://api.github.com/repos/opencontainers/runc/releases/latest | grep '"tag_name":' | cut -d '"' -f 4 | sed 's/v//')
+
+if [ -z "$RUNC_VERSION" ]; then
+    echo "Failed to fetch latest RUNC version. Using default version."
+    RUNC_VERSION="1.2.6"
+fi
+
+RUNC_URL="https://github.com/opencontainers/runc/releases/download/v${RUNC_VERSION}/runc.${ARCH}"
+RUNC_SHA256_URL="https://github.com/opencontainers/runc/releases/download/v${RUNC_VERSION}/SHA256SUMS"
+
+echo "Downloading runc ${RUNC_VERSION} from: $RUNC_URL"
+
+# Download runc binary and checksum
+curl -fsSL -o runc.${ARCH} ${RUNC_URL}
+curl -fsSL -o SHA256SUMS ${RUNC_SHA256_URL}
+
+# Verify SHA256 checksum
+EXPECTED_CHECKSUM=$(grep "runc.${ARCH}" SHA256SUMS | awk '{print $1}')
+ACTUAL_CHECKSUM=$(sha256sum runc.${ARCH} | awk '{print $1}')
+
+if [ "$EXPECTED_CHECKSUM" != "$ACTUAL_CHECKSUM" ]; then
+    echo "Checksum verification failed for runc. Exiting."
+    exit 1
+fi
+
+sudo install -m 755 runc.${ARCH} /usr/local/sbin/runc
+rm -f runc.${ARCH} SHA256SUMS
+
+echo "Runc ${RUNC_VERSION} installed successfully."
+
+# Install CNI plugins
+CNI_VERSION="1.1.1"
+CNI_TAR="cni-plugins-linux-${ARCH}-v${CNI_VERSION}.tgz"
+CNI_URL="https://github.com/containernetworking/plugins/releases/download/v${CNI_VERSION}/${CNI_TAR}"
+CNI_SHA256_URL="https://github.com/containernetworking/plugins/releases/download/v${CNI_VERSION}/SHA256SUMS"
+
+echo "Downloading CNI plugins from: $CNI_URL"
+
+# Download CNI tarball and checksum
+curl -fsSL -o ${CNI_TAR} ${CNI_URL}
+curl -fsSL -o SHA256SUMS ${CNI_SHA256_URL}
+
+# Verify SHA256 checksum
+EXPECTED_CHECKSUM=$(grep ${CNI_TAR} SHA256SUMS | awk '{print $1}')
+ACTUAL_CHECKSUM=$(sha256sum ${CNI_TAR} | awk '{print $1}')
+
+if [ "$EXPECTED_CHECKSUM" != "$ACTUAL_CHECKSUM" ]; then
+    echo "Checksum verification failed for CNI plugins. Exiting."
+    exit 1
+fi
+
+sudo mkdir -p /opt/cni/bin
+sudo tar Cxzvf /opt/cni/bin ${CNI_TAR}
+rm -f ${CNI_TAR} SHA256SUMS
+
+# Configure containerd
+sudo mkdir -p /etc/containerd
 containerd config default | sudo tee /etc/containerd/config.toml
```

Comment:
Question: Does `config default` generate a "sane" config with respect to the `SystemdCgroup`?